src/Mod/BIM/bimcommands/BimWindow.py

```diff
@@ -76,86 +104,105 @@ def Activated(self):
-                                host = obj.AttachmentSupport
-                            obj.AttachmentSupport = None # remove
-                    elif Draft.isClone(obj,"Window"):
-                        if obj.Objects[0].Inlist:
-                            host = obj.Objects[0].Inlist[0]
-
-                    FreeCAD.ActiveDocument.openTransaction(translate("Arch","Create Window"))
-                    FreeCADGui.addModule("Arch")
-                    FreeCADGui.doCommand("win = Arch.makeWindow(FreeCAD.ActiveDocument."+obj.Name+")")
-                    if host and self.Include:
-                        FreeCADGui.doCommand("win.Hosts = [FreeCAD.ActiveDocument."+host.Name+"]")
-                        siblings = host.Proxy.getSiblings(host)
-                        sibs = [host]
-                        for sibling in siblings:
-                            if not sibling in sibs:
-                                sibs.append(sibling)
-                                FreeCADGui.doCommand("win.Hosts = win.Hosts+[FreeCAD.ActiveDocument."+sibling.Name+"]")
-                    FreeCAD.ActiveDocument.commitTransaction()
-                    FreeCAD.ActiveDocument.recompute()
-                    return
-
-                # Try to detect an object to use as a window type - TODO we must make this safer
-
-                elif obj.Shape.Solids and (Draft.getType(obj) not in ["Wall","Structure","Roof"]):
-                    # we consider the selected object as a type
-                    FreeCAD.ActiveDocument.openTransaction(translate("Arch","Create Window"))
-                    FreeCADGui.addModule("Arch")
-                    FreeCADGui.doCommand("Arch.makeWindow(FreeCAD.ActiveDocument."+obj.Name+")")
-                    FreeCAD.ActiveDocument.commitTransaction()
-                    FreeCAD.ActiveDocument.recompute()
-                    return
-
-        # interactive mode
+        self.sel = FreeCADGui.Selection.getSelection()
+
+        if self.sel and self.handleSelectionMode():
+            return
+
+        # Fallback to interactive mode
+        self.handleInteractiveMode()
+
+    def handleSelectionMode(self):
+        import Draft
+
+        FreeCADGui.draftToolBar.offUi()
+        obj = self.sel[0]
+        if not hasattr(obj,'Shape'):
+            return False
+
+        obj_only_has_wires = obj.Shape.Wires and (not obj.Shape.Solids) and (not obj.Shape.Shells)
+        if obj_only_has_wires:
+            FreeCADGui.Control.closeDialog()
+
+            host = None
+            if hasattr(obj,"AttachmentSupport"):
+                host = get_attachment_object(obj)
+            elif Draft.isClone(obj,"Window") and obj.Objects[0].Inlist:
+                host = obj.Objects[0].Inlist[0]
+
+            FreeCAD.ActiveDocument.openTransaction(translate("Arch","Create Window"))
+            FreeCADGui.addModule("Arch")
+            FreeCADGui.doCommand("win = Arch.makeWindow(FreeCAD.ActiveDocument."+obj.Name+")")
```

Comment:
```suggestion
            FreeCADGui.doCommand(f"win = Arch.makeWindow(FreeCAD.ActiveDocument.{obj.Name})")
```
In general, would these `doCommand()` calls not be more readable like this? Or do f-strings not work in this case?